crates/pop-cli/src/common/wallet.rs

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::{
+	cli::traits::Cli,
+	wallet_integration::{FrontendFromString, TransactionData, WalletIntegrationManager},
+};
+use cliclack::{log, spinner};
+
+/// The prompt to ask the user if they want to use the wallet for signing.
+pub const USE_WALLET_PROMPT: &str = "Do you want to use your browser wallet to sign the extrinsic? (Selecting 'No' will prompt you to manually enter the secret key URI for signing, e.g., '//Alice')";
+
+/// Launches the wallet integration for in-browser signing. Blocks until the signature is received.
+///
+/// # Arguments
+/// * `call_data` - The call data to be signed.
+/// * `url` - Chain rpc.
+/// # Returns
+/// * The signed payload, if it exists.
+pub async fn request_signature(call_data: Vec<u8>, rpc: String) -> anyhow::Result<Option<String>> {
+	let ui = FrontendFromString::new(include_str!("../assets/index.html").to_string());
+
+	let transaction_data = TransactionData::new(rpc, call_data);
+	// Starts server with port 9090.
+	let mut wallet = WalletIntegrationManager::new(ui, transaction_data, Some(9090));
```

Comment:
If 9090 is taken, it will choose a random port. The UX is almost exactly the same. The only difference is that the last wallet used isn't saved between different ports (different URLs)